readthedocs/search/tasks.py

```diff
@@ -142,19 +142,45 @@ def delete_old_search_queries_from_db():
 
 
 @app.task(queue='web')
-def record_search_query(project_slug, version_slug, query, total_results):
-    """Record search query in database."""
-    if not project_slug or not version_slug or not query or not total_results:
+def record_search_query(project_slug, version_slug, query, total_results, time):
+    """Record/update search query in database."""
+    if not project_slug or not version_slug or not query:
         log.debug(
             'Not recording the search query. Passed arguments: '
-            'project_slug: %s, version_slug: %s, query: %s, total_results: %s' % (
-                project_slug, version_slug, query, total_results
+            'project_slug: %s, version_slug: %s, query: %s, total_results: %s, time: %s' % (
+                project_slug, version_slug, query, total_results, time
             )
         )
         return
 
-    project_qs = Project.objects.filter(slug=project_slug)
+    before_10_sec = time - timezone.timedelta(seconds=10)
+    partial_query_qs = SearchQuery.objects.filter(
+        project__slug=project_slug,
+        version__slug=version_slug,
+        query__startswith=query,
+        created__gte=before_10_sec,
+    ).order_by('-created')
+
+    # if partial query exists,
+    # just update it instead of creating new SearchQuery object.
+    if partial_query_qs.exists():
+        obj = partial_query_qs.first()
+        obj.created = time
+        obj.query = query
+        obj.save()
+        return
+
+    # don't record query with zero results.
+    if not total_results:
```

Comment:
Is there a reason this block isn't at the top of the function? I guess you could update the time on another query even if this one returns no results. However, that probably warrants a comment or something.

Reply:
> Is there a reason this block isn't at the top of the function?

Yes. I wanted to update the timing and query even if no results are there. For example: `readthed` will return no results but I wanted to update the timing and query if there is a previous query of `readth`.

> I guess you could update the time on another query even if this one returns no results. However, that probably warrants a comment or something.

I am not sure that I followed.